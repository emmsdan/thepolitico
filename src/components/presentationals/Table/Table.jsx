import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

// import logo from '../../../assets/images/official/favicon.png';

const Table = ({ header, content }) => {
  const items = content.map((item, key) => {
    return (
      <tbody key={key}>
        <tr>
          <td>
            {item.logo ? <img src={item.logo} className="circle" /> : item.id}
          </td>
          <td> {item.title} </td>
          <td> {item.description} </td>
          <td>
            {item.logo && (
              <Link to={`parties/${item.id}/edit`} className="button blue">
                edit
              </Link>
            )}
            <Link
              to="#delete"
              onClick={() => {
                alert(item.id);
              }}
              className="button red"
            >
              delete
            </Link>
          </td>
        </tr>
      </tbody>
    );
  });
  return (
    <table className="party-admin">
      <tbody>
        <tr>
          <th>{header.one}</th>
          <th>{header.two}</th>
          <th>{header.three}</th>
          <th>action</th>
        </tr>
      </tbody>
      {items}
    </table>
  );
};

Table.propTypes = {
  header: PropTypes.object,
  content: PropTypes.array,
};

export default Table;
