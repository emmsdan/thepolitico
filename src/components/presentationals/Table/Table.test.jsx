import React from 'react';
import { shallow } from 'enzyme';

import Table from './Table';

const table = {
  header: {
    one: 'Logo',
    two: 'party name',
    three: 'HEAD QUARTER ADDRESS',
  },
  content: [
    {
      id: 1,
      logo: 'party.logourl',
      title: 'party name',
      description: 'party hqaddress',
    },
  ],
};

describe('<Table />', () => {
  it('should render correctly', () => {
    const component = shallow(<Table {...table} />);
    expect(component).toMatchSnapshot();
  });
});
