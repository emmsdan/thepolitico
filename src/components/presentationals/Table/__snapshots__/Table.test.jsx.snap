// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render correctly 1`] = `
<table
  className="party-admin"
>
  <tbody>
    <tr>
      <th>
        Logo
      </th>
      <th>
        party name
      </th>
      <th>
        HEAD QUARTER ADDRESS
      </th>
      <th>
        action
      </th>
    </tr>
  </tbody>
  <tbody
    key="0"
  >
    <tr>
      <td>
        <img
          className="circle"
          src="party.logourl"
        />
      </td>
      <td>
         
        party name
         
      </td>
      <td>
         
        party hqaddress
         
      </td>
      <td>
        <Link
          className="button blue"
          to="parties/1/edit"
        >
          edit
        </Link>
        <Link
          className="button red"
          onClick={[Function]}
          to="#delete"
        >
          delete
        </Link>
      </td>
    </tr>
  </tbody>
</table>
`;
